dinstaller-lib/src/profile.rs

```diff
@@ -0,0 +1,125 @@
+use crate::manager::ManagerClient;
+use curl::easy::Easy;
+use jsonschema::JSONSchema;
+use serde_json;
+use std::{
+    error::Error,
+    fs, io,
+    io::{stdout, Write},
+    path::Path,
+    process::Command,
+};
+use tempfile::tempdir;
+
+/// Downloads a file a writes it to the stdout()
+///
+/// TODO: move this code to a struct
+/// TODO: add support for YaST-specific URLs
+/// TODO: do not write to stdout, but to something implementing the Write trait
+pub fn download(url: &str) -> Result<(), Box<dyn Error>> {
+    let mut easy = Easy::new();
+    easy.url(url)?;
+    easy.write_function(|data| {
+        stdout().write_all(data).unwrap();
```

Comment:
I would also add here TODO to retry download, ideally at least three times for less stable networks.